server/ServerFSM.cpp

```diff
@@ -891,7 +893,29 @@ sc::result MPLobby::react(const LobbyUpdate& msg) {
 
             // directly configurable lobby data
             m_lobby_data->m_new_game       = incoming_lobby_data.m_new_game;
-            m_lobby_data->m_players        = incoming_lobby_data.m_players;
+
+            int ai_count = 0;
+            int human_count = 0;
+            for (const auto& plr : incoming_lobby_data.m_players) {
+                switch (plr.second.m_client_type) {
+                case Networking::CLIENT_TYPE_AI_PLAYER:
+                    ai_count++;
+                    break;
+                case Networking::CLIENT_TYPE_HUMAN_PLAYER:
+                    human_count++;
+                    break;
+                }
+            }
+
+            // restrict count of AI
+            if (ai_count <= GetOptionsDB().Get<int>("mplobby-max-ai") || GetOptionsDB().Get<int>("mplobby-max-ai") < 0) {
+                m_lobby_data->m_players    = incoming_lobby_data.m_players;
+            } else {
+                has_important_changes = true;
+            }
+            if (human_count < GetOptionsDB().Get<int>("mplobby-min-human")) {
```

Comment:
add similar comment here about restricting count of humans

Reply:
Fixed.